test/traits.jl

```diff
@@ -0,0 +1,620 @@
+
+    # Ambiguous storage order, choose AlphaColor or ColorAlpha
+    @test_throws ErrorException ccolor(TransparentColor, HSV{Float32})
+    @test_throws ErrorException ccolor(TransparentColor{RGB}, HSV{Float32})
+    @test_throws ErrorException ccolor(TransparentColor{RGB,Float64}, HSV{Float32})
+    @test_throws ErrorException ccolor(TransparentColor{RGB{Float64},Float64}, HSV{Float32})
+    @test_throws ErrorException ccolor(TransparentColor{RGB{Float64},Float64,4}, HSV{Float32})
+
+    @test @inferred(ccolor(AlphaColor, RGB)) === ARGB
+    @test_broken @inferred(ccolor(AlphaColor, RGB{N0f8})) === ARGB{N0f8}
+    @test @inferred(ccolor(AlphaColor, RGB24)) === ARGB32
+    @test_broken @inferred(ccolor(AbstractARGB, RGB)) === ARGB
+    @test_broken @inferred(ccolor(AbstractARGB, RGB{N0f8})) === ARGB{N0f8}
+    @test_broken @inferred(ccolor(AbstractARGB, RGB24)) === ARGB32
+
+    for C in filter(T -> T <: AbstractRGB, ColorTypes.parametric3)
+        @test @inferred(ccolor(RGB24, C)) === RGB24
+        @test @inferred(ccolor(ARGB32, C)) === ARGB32
+        @test @inferred(ccolor(Gray24, C)) === Gray24
+        @test @inferred(ccolor(AGray32, C)) === AGray32
+    end
+
+    @test_throws MethodError ccolor(RGB, RGB{N0f8}(1,0,0))
+end
+
+# TODO: move `reinterpret` to `conversions.jl`
+@testset "reinterpret" begin
+    @test reinterpret(UInt32, RGB24()) === 0x00000000
+    @test reinterpret(UInt32, ARGB32()) === 0xff000000
+    @test reinterpret(UInt32, Gray24()) === 0x00000000
+    @test reinterpret(UInt32, AGray32()) === 0xff000000
+
+    @test reinterpret(UInt32, RGB24(0.071,0.204,0.337)) === 0x00123456
+    @test reinterpret(UInt32, ARGB32(0.071,0.204,0.337,0.671)) === 0xab123456
+    @test reinterpret(UInt32, Gray24(0.071)) === 0x00121212
+    @test reinterpret(UInt32, AGray32(0.071, 0.671)) === 0xab121212
+
+    @test reinterpret(RGB24, 0x00123456) === RGB24(0.071,0.204,0.337)
+    @test reinterpret(RGB24, 0xdc123456) ==  RGB24(0.071,0.204,0.337)
+    @test reinterpret(RGB24, 0x00123456) !== reinterpret(RGB24, 0xdc123456)
+    @test reinterpret(UInt32, reinterpret(RGB24, 0xdc123456)) === 0xdc123456
+
+    @test reinterpret(ARGB32, 0xab123456) === ARGB32(0.071,0.204,0.337,0.671)
+
+    @test reinterpret(Gray24, 0x00121212) === Gray24(0.071)
+    @test reinterpret(Gray24, 0xdc121212) ==  Gray24(0.071)
+    @test reinterpret(Gray24, 0x00121212) !== reinterpret(Gray24, 0xdc121212)
+    @test reinterpret(UInt32, reinterpret(Gray24, 0xdc121212)) === 0xdc121212
+    # the following behavior is undefined, so this is just for the regression test.
+    @test reinterpret(Gray24, 0xdc00ff12) ==  Gray24(0.071)
+    @test reinterpret(UInt32, reinterpret(Gray24, 0xdc00ff12)) === 0xdc00ff12
+
+    @test reinterpret(AGray32, 0xab121212) === AGray32(0.071, 0.671)
+
+    @test_throws MethodError reinterpret(UInt32, ARGB{N0f8}())
+    @test_throws ErrorException reinterpret(ARGB{N0f8}, 0x12345678)
+end
+
+@testset "comparisons" begin
```

Comment:
From here I'd say this is less `traits` and more `operations`

Reply:
I agree with you. However I categorized the tests based on the locations of target functions.

The `==` and `≈` are definitely "operators", but, I think, `isapprox` is something like "traits". Although both definitions should be close, I think they should be placed before "src/operations.jl" because of their primitivity.
https://github.com/JuliaGraphics/ColorTypes.jl/blob/2c6bd3451988a974a1008c61a58dcd599a754553/src/ColorTypes.jl#L43-L47
So, what about moving them later to a new file "comparison.jl"?

Reply:
A similar problem exists with `alphacolor`/`coloralpha`. However, since both "for types" are needed to define the constructors, I think they should still be in "src/types.jl".